_posts/2019-02-05-configurable-builds-part-1.md

```diff
@@ -0,0 +1,280 @@
+---
+layout: posts
+title: Configurable Builds - Part 1
+authors:
+ - gregestren
+---
+
+The larger your project gets, the more likely you have to build it in different
+ways.
+
+Maybe you need Android, iOS, and desktop versions of your app. Maybe you build
+C++ for different platforms. Or maybe you maintain a popular library that
+different users need different features from.
+
+These are examples of *configuration*: the process of building the same *code*
+with different *settings* to customize it for specific needs. In Bazel-speak,
+this means building a
+[target](https://docs.bazel.build/versions/master/build-ref.html#targets) with
+a set of
+[flags](https://docs.bazel.build/versions/master/user-manual.html#flags-options):
+
+```sh
+$ bazel build //my:cc_binary --cpu=x86
+$ bazel build //my:cc_binary --cpu=arm
+$ bazel build //my:android_binary --android_sdk=@androidsdk//:sdk-28
+```
+
+This post reviews Bazel's *configurability work*, which is an ongoing effort to
+to make these tasks simple, flexible, and powerful. This is a deeper dive into
+themes covered in Bazel's [configurability
+roadmap](https://bazel.build/roadmaps/configuration.html).
+
+## History
+
+Bazel's a
+[powerful](https://docs.bazel.build/versions/master/bazel-vision.html) build
+tool that's especially suited for large,
+[multi-language](/2018/12/05/multilanguage-build-system.html) builds. But it
```

Comment:
This link is broken when I try it.

Reply:
Re: link: it's a relative link to blog.bazel so it works when published. But sure, you can read it in GitHub too - changed to the full URL.

Re: clarifying: done and agreed.